owner/src/main/java/org/the_chance/honeymart/ui/add_product/AddProductViewModel.kt

```diff
@@ -19,12 +21,8 @@ class AddProductViewModel @Inject constructor(
         _state.update { it.copy(isLoading = true) }
         tryToExecute(
             {
-                addProductWithImagesUseCase(
-                    product.name,
-                    product.price.toDouble(),
-                    product.description,
-                    categoryId,
-                    product.images
+                addProductUseCase(
+                    product.name, product.price.toDouble(), product.description, categoryId,
                 ).toAddProductUiState()
```

Comment:
move the mapper "toAddProductUiState()" from tryToExecute() to onAddProductSuccess() funtion.

Reply:
your solution is wrong, that's give error because the use case return entity and in view model you must mapping the use case to ui state